framework/db/ConstraintFinderTrait.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\db;
+
+/**
+ * ConstraintFinderTrait provides methods for getting a table constraint information.
+ *
+ * @property CheckConstraint[][] $schemaChecks Check constraints for all tables in the database.
+ * Each array element is an array of [[CheckConstraint]] or its child classes. This property is read-only.
+ * @property DefaultValueConstraint[] $schemaDefaultValues Default value constraints for all tables in the database.
+ * Each array element is an array of [[DefaultValueConstraint]] or its child classes. This property is read-only.
+ * @property ForeignKeyConstraint[][] $schemaForeignKeys Foreign keys for all tables in the database. Each
+ * array element is an array of [[ForeignKeyConstraint]] or its child classes. This property is read-only.
+ * @property IndexConstraint[][] $schemaIndexes Indexes for all tables in the database. Each array element is
+ * an array of [[IndexConstraint]] or its child classes. This property is read-only.
+ * @property Constraint[] $schemaPrimaryKeys Primary keys for all tables in the database. Each array element
+ * is an instance of [[Constraint]] or its child class. This property is read-only.
+ * @property IndexConstraint[][] $schemaUniques Unique constraints for all tables in the database.
+ * Each array element is an array of [[IndexConstraint]] or its child classes. This property is read-only.
+ *
+ * @author Sergey Makinen <sergey@makinen.ru>
+ * @since 2.0.13
+ */
+trait ConstraintFinderTrait
```

Comment:
Do you mean by replacing abstracts with `NotSupportedException` (to maintain BC)? It's possible. The only reason behind this trait was to possibly make `Schema` look "lighter" if someone doesn't need a constraint support at all.
Maybe it's just an overcomplication and we simply need to put them all in `Schema`. I even suggested the same for `ViewFinderTrait`… I don't know TBH. :)
cc @samdark